pyincore_data/utils/nsiutil.py

```diff
@@ -0,0 +1,247 @@
+        script_dir = os.path.dirname(os.path.abspath(__file__))
+        csv_dir = os.path.join(script_dir, 'data', 'nsi', 'occ_bldg_mapping', region.lower())
+
+        o2b_dict = {}  # Dictionary to store occupancy-to-building type mappings
+
+        # Iterate over all CSV files in the directory
+        for filename in os.listdir(csv_dir):
+            if filename.endswith(".csv"):
+                sheet_name = filename.replace(".csv", "")  # Extract the original sheet name
+                file_path = os.path.join(csv_dir, filename)
+
+                # Read CSV file
+                df = pd.read_csv(file_path)
+
+                # Clean 'OccClass' column
+                df['OccClass'] = df['OccClass'].apply(lambda x: str(x).replace(u'\xa0', u''))
+                df.set_index('OccClass', inplace=True)
+
+                # Clean column names
+                old_cols = list(df.columns)
+                new_cols = [str(x).replace(u'\xa0', u'') for x in old_cols]
+                rename_dict = dict(zip(old_cols, new_cols))
+                df.rename(columns=rename_dict, inplace=True)
+
+                o2b_dict[sheet_name] = df
+
+        return o2b_dict
+
+    @staticmethod
+    def assign_hazus_specific_structure_type(gdf, region, sensitivity_analysis=False, random=False):
+        """
+        Function to map HAZUS-specific occupancy types to HAZUS-specific building types.
+        Adjusts logic based on the region (WestCoast, MidWest, EastCoast).
+
+        Contributors
+        | Original Code and Logic: Dylan R. Sanderson
+        | Implementation: Yong Wook Kim
+
+        Inputs:
+            - gdf: GeoDataFrame containing NSI data.
+            - region (str): The region name.
+            - sensitivity_analysis (bool): If True, applies sensitivity analysis.
+            - random (bool): If True, selects building type randomly based on probability distribution.
+
+        Returns:
+            GeoDataFrame with additional columns.
+        """
+        np.random.seed(1337)
+
+        o2b_dict = NsiUtil.read_occ_building_mapping(region)
+
+        guid = []
+        struct_typ = []
+        no_stories = []
+        year_built = []
+        dgn_lvl = []
+        exact_match = []
+        fallback_count = 0
+        total_records = len(gdf)
+
+        cnt_nan = 0
+        for i, row in gdf.iterrows():
+            guid.append(str(uuid.uuid4()))
+            year_built_ = row['med_yr_blt']
+            no_stories_ = row['num_story']
+            occ_type_ = row['occtype'].split('-')[0]
+            exact_match_flag = "Yes"
+
+            if "RES3" in occ_type_:
+                occ_type_ = "RES3"
+
+            if occ_type_ == 'RES1':
+                struct_typ.append("W1")
+                no_stories.append(no_stories_)
+                year_built.append(year_built_)
+                dgn_lvl.append(NsiUtil.year_built_to_dgn_lvl(year_built_))
+                exact_match.append(exact_match_flag)
+                continue
```

Comment:
This continue will cause the skipping of the `region == "WestCoast" logic`, is that intended?

Reply:
yes